### PR TITLE
re-architecture and adding info on type changes

### LINKED ISSUE #5016: Docs + Upgrade guide: Other types changes
https://github.com/prisma/prisma-engines/pull/3969
https://github.com/prisma/prisma-engines/pull/3967

### PR DESCRIPTION
**Pull request by jharrell**
Resolves #5016

## Describe this PR

- moved Prisma 5 guide to be directory-based
- made jsonProtocol changes a sub page
- added missing information on type changes

### CODE REVIEW

**Comment by jharrell** [comment]
@janpio let me know what you think about the switch to a directory. I'm also looking at making a subpage for `rejectOnNotFound` changes.

**Comment by janpio** [comment]
I like the idea.

- Can we remove the "in Prisma 5" from the left hand sidebar? That is repeated info - in the page title it might be useful though.
- `jsonProtocol` should always be code
- Instead of `For further information on what changes ...` you could try to summarize the changes, so it is clearer what the use can expect there.

**Comment by jharrell** [comment]
@janpio updated the `jsonProtocol` references and added a sub-page for `rejectOnNotFound`.

File: content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5/101-jsonprotocol-changes.mdx [hunk 1]
```diff
@@ -0,0 +1,336 @@
+---
+title: 'jsonProtocol changes'
+metaTitle: 'Upgrade to jsonProtocol in Prisma 5'
+metaDescription: 'Guides on how to upgrade to jsonProtocol in Prisma 5'
```
Comment (janpio) [inline content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5/101-jsonprotocol-changes.mdx]: I think this is misleading right now.

**Review by janpio — APPROVED** [review]
👍 

(Please also get reviews from DA, I am semi blind on that topic now and do not pick up things any more)

**Review by Weakky — APPROVED** [review]
Approving the "Generated type changes" section 👍

**Pull request merged**